gc.c

```diff
@@ -2759,11 +2759,20 @@ rb_class_allocate_instance(VALUE klass)
     return obj;
 }
 
+static inline void
+rb_data_object_check(VALUE klass)
+{
+    Check_Type(klass, T_CLASS);
```

Comment:
```suggestion
```

`rb_get_alloc_func` has checked if `klass` is `T_CLASS`, and `rb_cObject` is `T_CLASS` of course.

Reply:
Of course! I will remove this explicit check. Thank you.